v12.16.1 proposal (#31781)
nodejs/node: doc/changelogs/CHANGELOG_V12.md

  * [io.js](CHANGELOG_IOJS.md)
  * [Archive](CHANGELOG_ARCHIVE.md)

<a id="12.16.1"></a>
## 2020-02-13, Version 12.16.1 'Erbium' (LTS), @MylesBorins

### Notable changes

> Node.js 12.16.0 included large update to the ESM implementation in the 12.x release stream.

Reviewer: ```suggestion
Node.js 12.16.0 included a large update to the ESM implementation.
```

> One of the new features, Self Referential Modules, accidentally was released not behind the

Reviewer: ```suggestion
One of the new features, Self Referential Modules, was accidentally released without requiring the
```

Author: thanks rich. I'll make sure this is also reflected in the commit message

> `--experimental-modules` flag. This release is being made specifically to appropraitely flag

Reviewer: ```suggestion
`--experimental-modules` flag. This release is being made to appropriately flag
```

> the feature which is not yet ready to be release on LTS.

Reviewer: ```suggestion
the feature.
```

A Semver-Minor change to introduce `--use-largepages` as a runtime option
introduced a linking failure. This had been fixed in master but regressed as the fix has not yet gone out
in a Current release. The feature has been reverted, but will be able to reland with a fix in a future
Semver-Minor release.

**Async Hooks was Causing an Exception When Handling Errors**:

Changes in async hooks internals introduced a case where an internal api call could be called with undefined
causing a process to crash. The change to async hooks was reverted. A regression test and fix has been proposed
and the change could re land in a future Semver-Patch release if the regression is reliably fixed.

**New Enumerable Read-Only Property on EventEmitter breaks @types/extend**

A new property for enumerating events was added to the EventEmitter class. This
broke existing code that was using the `@types/extend` module for extending classses
as `@types/extend` was attemping to write over the existing field which the new
change made read-only. As this is the first property on EventEmitter that is
read-only this feature could be considered Semver-Major. The new feature has been

> reverted but could re land in a future Semver-Minor release if a non breaking way

Reviewer 2: ```suggestion
reverted but could re-land in a future Semver-Minor release if a non breaking way
```

of applying it is found.

**Exceptions in the HTTP parser were not emitting an uncaughtException**

A refactoring to Node.js interanls resulted in a bug where errors in the HTTP

> parser were not being emitted by `process.on('uncaughtException')`. The fix

Reviewer 2: Might be good to mention that this only applies in situations where an async_hooks `after` hook exists

(Also, typo: internals)
